### PR TITLE
explorer: progress indicator

### PR DESCRIPTION
**Pull request by mike-hunhoff**
Addresses #300.

Features:

* display progress for rule loading and feature extraction
* user can cancel analysis at any time

Bug fixes:

* reset IDA highlighting before program rebase
* reset IDA highlighting before new analysis is performed
* only restart capa analysis if image base actually changes when user elects to change program base

### CODE REVIEW

**Review by mr-tz — APPROVED** [review]

File: capa/ida/plugin/form.py [hunk 1]
```diff
@@ -30,6 +31,47 @@
...
+    """
+
+    def __init__(self):
+        super(CapaExplorerFeatureExtractor, self).__init__()
+        self.indicator = CapaExplorerProgressIndicator()
+
+    def extract_function_features(self, f):
+        self.indicator.update("function at 0x%X" % f.start_ea)
+        for (feature, ea) in capa.features.extractors.ida.function.extract_features(f):
+            yield feature, ea
```
Comment (williballenthin) [inline capa/ida/plugin/form.py:72]: this is a clever way to implement progress, nice job!

File: capa/ida/plugin/form.py [hunk 2]
```diff
@@ -30,6 +31,47 @@
...
+    track progress during feature extraction, also allow user to cancel feature extraction
+    """
+
+    def __init__(self):
+        super(CapaExplorerFeatureExtractor, self).__init__()
+        self.indicator = CapaExplorerProgressIndicator()
+
+    def extract_function_features(self, f):
+        self.indicator.update("function at 0x%X" % f.start_ea)
+        for (feature, ea) in capa.features.extractors.ida.function.extract_features(f):
```
Comment (williballenthin) [inline capa/ida/plugin/form.py:71]: should we call the super impl here or just reimplement?

**Review by williballenthin — APPROVED** [review]
nice!

File: capa/ida/plugin/form.py (hunk 2 above)
Comment (mike-hunhoff) [reply capa/ida/plugin/form.py]: absolutely! updated to use super impl

**Pull request merged**